set_user--1.0.sql

```diff
@@ -23,6 +23,17 @@ RETURNS text
 AS 'MODULE_PATHNAME', 'set_user'
 LANGUAGE C STRICT;
 
+CREATE FUNCTION set_user_u(text)
+RETURNS text
+AS 'MODULE_PATHNAME', 'set_user'
+LANGUAGE C;
+
+CREATE FUNCTION set_user_u(text, text)
```

Comment:
Does it make sense to have a two arg version of the superuser escalation function? The second arg would be a token controlling reset, but we are already superuser. Additionally your check for funcname only occurs when nargs == 1.